src/main/java/org/dependencytrack/tasks/scanners/TrivyAnalysisTask.java

```diff
@@ -0,0 +1,553 @@
+            final CloseableHttpResponse response = RETRY.executeCheckedSupplier(() -> HttpClientPool.getClient().execute(request));
+            LOGGER.info("DeleteBlob response: " + response.getStatusLine().getStatusCode());
+        } catch (Throwable  ex) {
+            handleRequestException(LOGGER, ex);
+        }
+
+    }
+
+    private void handle(final Component component, final org.dependencytrack.parser.trivy.model.Vulnerability data) {
+        if (component == null) {
+            LOGGER.error("Unable to handle null component");
+            return;
+        } else if (data == null) {
+            addNoVulnerabilityToCache(component);
+            return;
+        }
+
+        try (QueryManager qm = new QueryManager()) {
+            final var trivyParser = new TrivyParser();
+
+            Vulnerability synchronizedVulnerability = trivyParser.parse(data, qm);
+            addVulnerabilityToCache(component, synchronizedVulnerability);
+            final Component componentPersisted = qm.getObjectByUuid(Component.class, component.getUuid());
+            if (componentPersisted != null && synchronizedVulnerability.getVulnId() != null) {
+                NotificationUtil.analyzeNotificationCriteria(qm, synchronizedVulnerability, componentPersisted, vulnerabilityAnalysisLevel);
+                qm.addVulnerability(synchronizedVulnerability, componentPersisted, this.getAnalyzerIdentity());
+                LOGGER.info("Trivy vulnerability added : " + synchronizedVulnerability.getVulnId() + " to component " + component.getName());
+            }
+            Event.dispatch(new IndexEvent(IndexEvent.Action.COMMIT, Vulnerability.class));
+
+            updateAnalysisCacheStats(qm, Vulnerability.Source.NVD, apiBaseUrl, component.getPurl().getCoordinates(), component.getCacheResult());
+        }
+    }
+
+
+
+    private Optional<String> getApiBaseUrl() {
+        if (apiBaseUrl != null) {
+            return Optional.of(apiBaseUrl);
+        }
+
+        try (final var qm = new QueryManager()) {
+            final ConfigProperty property = qm.getConfigProperty(
+                    ConfigPropertyConstants.SCANNER_TRIVY_BASE_URL.getGroupName(),
+                    ConfigPropertyConstants.SCANNER_TRIVY_BASE_URL.getPropertyName()
+            );
+            if (property == null) {
+                return Optional.empty();
+            }
+
+            apiBaseUrl = UrlUtil.normalize(property.getPropertyValue());
+            return Optional.of(apiBaseUrl);
+        }
+    }
+
+    private Supplier<String> createTokenSupplier(final String tokenValue) {
+        final String[] tokens = tokenValue.split(";");
+        if (tokens.length > 1) {
+            LOGGER.debug("Will use %d tokens in round robin" .formatted(tokens.length));
+            final var roundRobinAccessor = new RoundRobinAccessor<>(List.of(tokens));
+            return roundRobinAccessor::get;
+        }
+
+        return apiTokenSupplier = () -> tokenValue;
+    }
```

Comment:
Similar to the thread pool, this was merely a workaround for Snyk, as it has (had?) quite low rate limits. So we added the ability to use multiple tokens in order to raise the limit. This should not be a problem for Trivy, so this can be removed.